src/Custom/Microsoft.Extensions.AI/OpenAIResponseChatClient.cs

```diff
@@ -0,0 +1,603 @@
+                                callInfo.ResponseItem.CallId,
+                                callInfo.ResponseItem.FunctionName,
+                                static json => JsonSerializer.Deserialize(json, ResponseClientJsonContext.Default.IDictionaryStringObject)!);
+
+                            lastMessageId = callInfo.ResponseItem.Id;
+                            lastRole = ChatRole.Assistant;
+                            yield return new ChatResponseUpdate(lastRole, [fci])
+                            {
+                                CreatedAt = createdAt,
+                                MessageId = lastMessageId,
+                                ModelId = modelId,
+                                ResponseId = responseId,
+                                ChatThreadId = responseId,
+                            };
+                        }
+
+                        break;
+                    }
+
+                case StreamingResponseErrorUpdate errorUpdate:
+                    yield return new ChatResponseUpdate
+                    {
+                        CreatedAt = createdAt,
+                        MessageId = lastMessageId,
+                        ModelId = modelId,
+                        ResponseId = responseId,
+                        ChatThreadId = responseId,
+                        Contents =
+                        [
+                            new ErrorContent(errorUpdate.Message)
+                        {
+                            ErrorCode = errorUpdate.Code,
+                            Details = errorUpdate.Param,
+                        }
+                        ],
+                    };
+                    break;
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    void IDisposable.Dispose()
+    {
+        // Nothing to dispose. Implementation required for the IChatClient interface.
+    }
+
+    /// <summary>Creates a <see cref="ChatRole"/> from a <see cref="MessageRole"/>.</summary>
+    private static ChatRole ToChatRole(MessageRole? role) =>
+        role switch
+        {
+            MessageRole.System => ChatRole.System,
+            MessageRole.Developer => _chatRoleDeveloper,
+            MessageRole.User => ChatRole.User,
+            _ => ChatRole.Assistant,
+        };
+
+    /// <summary>Creates a <see cref="ChatFinishReason"/> from a <see cref="ResponseIncompleteStatusReason"/>.</summary>
+    private static ChatFinishReason? ToFinishReason(ResponseIncompleteStatusReason? statusReason) =>
+        statusReason == ResponseIncompleteStatusReason.ContentFilter ? ChatFinishReason.ContentFilter :
+        statusReason == ResponseIncompleteStatusReason.MaxOutputTokens ? ChatFinishReason.Length :
+        null;
+
+    /// <summary>Converts a <see cref="ChatOptions"/> to a <see cref="ResponseCreationOptions"/>.</summary>
+    private static ResponseCreationOptions ToOpenAIResponseCreationOptions(ChatOptions? options)
+    {
+        ResponseCreationOptions result = new();
```

Comment:
Same thing about `ResponseCreationOptions` as in the `OpenAIChatClient` equivalent.